api/pkg/util/image.go

```diff
@@ -0,0 +1,10 @@
+package util
+
+import (
+	"sigs.k8s.io/kustomize/api/internal/image"
+)
+
+// Splits image string name into name, tag and digest
+func SplitImageName(imageName string) (name string, tag string, digest string) {
```

Comment:
does this need to be in its own file and exposed publicly? Seems like it can just be a private helper in the file that it is used in

Reply:
This function calls `image.Split` [here](https://github.com/kubernetes-sigs/kustomize/pull/5234/files#diff-669bbf61eeb56134d2813eec14f367411350c932e610cc664a36bca43a856301R9). However, image is now private, so I cannot directly call it from the `setImage` command.

I believe this is what @varshaprasad96 proposed: to expose only this helper function rather than making the entire Image class public. Perhaps I misunderstood.

Reply:
I was looking into update the update.go file as well, but there's another line where the other function from the class `image` is being used. (`image.IsImageMatched`) 

https://github.com/kubernetes-sigs/kustomize/blob/779f1530719c61df642e69675c7c47c87e4a1267/api/filters/imagetag/updater.go#L30

If I expose `image.IsImageMatched` in the utils helper, the entire `Image` class would be expose as public. Then it wouldn't make sense to made it internal in the first place. 🤔 

Given that the `updater.go` `imagetag` package is part of `api`, we could just leave the cal to `split` to the internal class rather than calling it through the helper.

Reply:
@varshaprasad96 @natasha41575 this is polite ping :)

Reply:
@blackjid apologies for missing the ping. Looking at it again, the whole intention of making `image.Split` internal was for us not needing to support any helpers as APIs and have restrictions on breaking them. But exposing everything in util and then calling an internal method doesn't make sense. This is not a complicated API for us to support, and I would not want to block your PR on this. So imo we can just externalize it and of ahead to merge the PR.

Reply:
Ok, so.., I can put back `Image`

> let's place this under the same pkg as before `api/filter/imagetag`

I think before was in `api/image`.. 
https://github.com/kubernetes-sigs/kustomize/pull/5261/files

Do you mean revert your changes? or to take a different approach?

Reply:
@varshaprasad96 ?